### PR TITLE
Add layout component

### PR DESCRIPTION
**Pull request by jklina**
This promotes the new `Layout` component from `Experimental` to `Alpha`. 

I can do a fast follow of migrating this linter for upstreaming: https://github.com/github/github/pull/198781

### CODE REVIEW

File: docs/src/@primer/gatsby-theme-doctocat/nav.yml [hunk 1]
```diff
@@ -110,3 +110,5 @@
       url: /components/alpha/underlinenav
     - title: UnderlinePanels
       url: /components/alpha/underlinepanels
+    - title: Layout - Fix my order in docs/src/@primer/gatsby-theme-doctocat/nav.yml
```
Comment (camertron) [inline docs/src/@primer/gatsby-theme-doctocat/nav.yml:113]: Does this need to be addressed?

File: static/arguments.yml [hunk 2]
```diff
@@ -29,6 +29,28 @@
...
+  parameters:
+  - name: stacking_breakpoint
+    type: Symbol
+    default: "`:md`"
+    description: When the `Layout` should change from rows into columns. One of `:lg`,
+      `:md`, or `:sm`.
+  - name: first_in_source
+    type: Symbol
+    default: "`:sidebar`"
+    description: Which element to render first in the HTML. This will change the keyboard
```
Comment (camertron) [inline static/arguments.yml:43]: Shouldn't this have a "one of" part?

**Review by camertron — APPROVED** [review]

**Comment by camertron** [comment]
This looks great, thanks @jklina!

File: static/arguments.yml (hunk 2 above)
Comment (jklina) [reply static/arguments.yml:43]: Yep! https://github.com/primer/view_components/pull/889/commits/8782f1d5f8895d56f25542e072efc5338cba42b6

File: docs/src/@primer/gatsby-theme-doctocat/nav.yml (hunk 1 above)
Comment (jklina) [reply docs/src/@primer/gatsby-theme-doctocat/nav.yml]: Looks like it was a duplicate entry. Removed here: https://github.com/primer/view_components/pull/889/commits/02ae8af072c81513015c608ded005a7de761648b

Comment (jklina) [reply docs/src/@primer/gatsby-theme-doctocat/nav.yml]: Looks like it was a duplicate entry. Removed here: https://github.com/primer/view_components/pull/889/commits/02ae8af072c81513015c608ded005a7de761648b

File: static/arguments.yml (hunk 2 above)
Comment (jklina) [reply static/arguments.yml:43]: Yep! https://github.com/primer/view_components/pull/889/commits/8782f1d5f8895d56f25542e072efc5338cba42b6

**Review by camertron — APPROVED** [review]

**Pull request merged**